### ISSUE TITLE
error using load_lora_weights with dora

### ISSUE THREAD

**Issue by SlZeroth** [huggingface/diffusers#7366]
I got error from my trained xl dora file 

pipe = StableDiffusionXLPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    torch_dtype=torch.bfloat16
)

pipe.to("cuda")

pipe.load_lora_weights("e:/diffusion-models/Lora/test16/woman_young_64rank.safetensors")
pipe.fuse_lora(lora_scale=0.85)

got error. this is error information

<img width="1230" alt="image" src="https://github.com/huggingface/peft/assets/8098724/255dcdda-54c2-459f-aa0e-6e71af6b18d6">

**Comment by sayakpaul** [comment]
Transferred this issue from `peft` as I think it's better suited here.

**Comment by sayakpaul** [comment]
@SlZeroth does https://github.com/huggingface/diffusers/pull/7371 work for you?

**Comment by BenjaminBossan** [comment]
What's still unclear to me is where the `dora_scale` name comes from.

**Comment by sayakpaul** [comment]
`dora_scale` comes from the community [checkpoint](https://github.com/huggingface/diffusers/issues/7366#issuecomment-2002878393) (that doesn't use `peft` for training). `dora_scale` corresponds to `lora_magnitude_vector`.

**Comment by BenjaminBossan** [comment]
> `dora_scale` comes from the community [checkpoint](https://github.com/huggingface/diffusers/issues/7366#issuecomment-2002878393) (that doesn't use `peft` for training)

Thanks, but what I meant is where is the code coming from that created that checkpoint? Right now, the PR relies on the key name being `dora_scale`, but can we be certain of that? Is this some standard training script being used or are we trying to accomodate a custom script of unknown provenance?

**Comment by sayakpaul** [comment]
> Thanks, but what I meant is where is the code coming from that created that checkpoint?

It's coming from https://github.com/KohakuBlueleaf/LyCORIS/tree/dev/lycoris. We support the non-DoRA LoRAs trained using community trainers (LastBen, Kohya, etc.). These are quite popular in the diffusion community actually. 

> Right now, the PR relies on the key name being dora_scale, but can we be certain of that? Is this some standard training script being used or are we trying to accommodate a custom script of unknown provenance?

Yeah, it's a bit handwavy. In fact, if you look into `_convert_kohya_lora_to_diffusers()`, we rely on the provided state dict for determining if it's non-diffusers and currently, that's the best we have got. A significant usage comes from these LoRAs. https://huggingface.co/spaces/multimodalart/LoraTheExplorer gives a collection of assorted LoRAs that were trained using non-diffusers trainers and yield amazing quality. 

LMK if there's anything unclear

**Comment by SlZeroth** [comment]
I got same issue with diffusers trainer too ( advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py )

this is the command for train model with dora

accelerate launch --num_cpu_threads_per_process=2 ./train_dreambooth_lora_sdxl_advanced.py --pretrained_model_name_or_path="streamize/xxmixgirl" --instance_prompt="a photo of a ohwx woman" --validation_prompt="a photo of a ohwx woman in the home" --output_dir="lora-rank64-2000-20" --resolution=1024 --train_batch_size=1 --repeats=1 --gradient_accumulation_steps=1 --gradient_checkpointing --learning_rate=1.0 --text_encoder_lr=1.0 --optimizer="prodigy" --mixed_precision="bf16" --train_text_encoder_ti --train_text_encoder_ti_frac=0.5 --lr_scheduler=constant --lr_warmup_steps=0 --rank=64 --use_dora --max_train_steps=2000 --checkpointing_steps=2000 --seed=0 --instance_data_dir="C:/Users/qkrwn/OneDrive/face_images/bad_people/young"

I got this error when training 

Loading adapter weights from state_dict led to unexpected keys not found in the model:  ['down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_out.0.lora_magnitude_vector.default_0 ....
but the training is sucessfully finish and i got safetensor file

I m trying to inference with trained safetensors file

pipeline = StableDiffusionXLPipeline.from_pretrained("streamize/xxmixgirl", dtype=torch.bfloat16)
pipeline.to(device="cuda")

pipeline.scheduler = EulerAncestralDiscreteScheduler.from_config(pipeline.scheduler.config)
pipeline.enable_model_cpu_offload()
pipeline.enable_xformers_memory_efficient_attention()

state_dict = load_file("./lora-rank100-2000-20/lora-rank100-2000-20_emb.safetensors")
pipeline.load_textual_inversion(state_dict["clip_l"], token=["<s0>", "<s1>"], text_encoder=pipeline.text_encoder, tokenizer=pipeline.tokenizer)
# load embeddings of text_encoder 2 (CLIP ViT-G/14)
pipeline.load_textual_inversion(state_dict["clip_g"], token=["<s0>", "<s1>"], text_encoder=pipeline.text_encoder_2, tokenizer=pipeline.tokenizer_2)
pipeline.load_lora_weights("./lora-rank100-2000-20/pytorch_lora_weights.safetensors")
pipeline.fuse_lora(lora_scale=0.85)

images = pipeline(
    "xxmixgirl, photo of ohwx woman",
    negative_prompt="(worst quality, low quality, illustration, 3d, 2d, painting, cartoons, sketch), tooth, open mouth",
    num_inference_steps=25,
).images

images[0].save("xxmixgirl_test_image.png")

I got this warning message when inference

Loading adapter weights from state_dict led to unexpected keys not found in the model:  ['down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_k.lora_magnitude_vector.default_0', 'down_blocks.1...

and the inference image quality is bad 

how is the problem load the safetensors file?

this is the version of my packages

>>> import diffusers
>>> diffusers.__version__
'0.27.0'
>>> import peft
>>> peft.__version__
'0.9.0'

**Comment by sayakpaul** [comment]
Can we focus on the given issue first? You opened an issue asking if a community-trained DoRA LoRA could be supported. We opened a PR fixing that and asked if that was working for you. Would appreciate if you could provide a response [there](https://github.com/huggingface/diffusers/issues/7366#issuecomment-2003356725). 

And training related issues should be tackled in a separate issue thread.

**Comment by SlZeroth** [comment]
thank you I check this
